block-storage/src/test/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorageTest.scala

```diff
@@ -338,4 +339,127 @@ class BlockDagKeyValueStorageTest extends BlockDagStorageTest {
         _       = effects shouldBe Set(b1, b2, b3).map(_.blockHash)
       } yield ()
     }
+
+  /**
+    * Assuming recreating view of target block T.
+    * Bi are seen, 0i not seen, but already in the most recent view.
+    * All Os should be removed.
+    *
+    * O2 and O3 are high excess
+    * O1 is range excess
+    * O4 is unseenSender excess - some validator bonded in genesis has been waiting for a while before creating new block.
+    *   And created on top of genesis without referencing all new messages.
+    *
+    *   03
+    *   02     T
+    *   L1 01 L2
+    *      L0
+    *   B0 B1 B2 O4
```

Comment:
@tgrospic we've been talking about possibility of messages below lowest target LM to be outside the view. I still not sure this is impossible, O4 block is an example. Therefore I've added this check to truncate
https://github.com/rchain/rchain/blob/4725a25a9e89d73dc9aa0dfd03e488d332e77479/block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala#L145-L171

Reply:
So these messages should be with higher sequence number then LMs. They are like attached sub-dag created after LMs from which we want to truncate.

Maybe we can sequence number to cut off any newer messages. Traversing from LMs until oldest LM, collecting excess children, building part of new child map, building part of height map and unfinalized set.